packages/dds/sequence/src/test/revertibles.spec.ts

```diff
@@ -169,14 +283,187 @@ describe("Sequence.Revertibles with Remote Edits", () => {
+
+		collection2.change(id, 3, 8);
+
+		collection.on("changeInterval", (interval, previousInterval, local, op) => {
+			appendLocalChangeToRevertibles(sharedString, interval, previousInterval, revertibles);
+		});
+		collection.change(id, 4, 9);
+		containerRuntimeFactory.processOneMessage();
+
+		revertIntervalRevertibles(sharedString, revertibles.splice(0));
+		containerRuntimeFactory.processAllMessages();
+
+		assertIntervals(sharedString, collection, [{ start: 0, end: 5 }]);
+		assertIntervals(sharedString2, collection2, [{ start: 0, end: 5 }]);
+	});
+	it("remote interval change interacting with reverting an interval remove", () => {
+		sharedString.insertText(0, "hello world");
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		const id = collection.add(0, 5, IntervalType.Simple).getIntervalId()!;
+		containerRuntimeFactory.processAllMessages();
+
+		collection.on("deleteInterval", (interval, local, op) => {
+			appendLocalDeleteToRevertibles(sharedString, interval, revertibles);
+		});
+		collection.removeIntervalById(id);
+		containerRuntimeFactory.processOneMessage();
+
+		collection2.change(id, 3, 8);
+
+		revertIntervalRevertibles(sharedString, revertibles.splice(0));
+		containerRuntimeFactory.processAllMessages();
+
+		assertIntervals(sharedString, collection, [{ start: 0, end: 5 }]);
+		assertIntervals(sharedString2, collection2, [{ start: 0, end: 5 }]);
+	});
+	it("remote interval remove interacting with reverting an interval property change", () => {
+		sharedString.insertText(0, "hello world");
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		const id = collection.add(0, 5, IntervalType.Simple, { foo: "one" }).getIntervalId()!;
+		containerRuntimeFactory.processAllMessages();
+
+		collection.on("propertyChanged", (interval, propertyDeltas, local, op) => {
+			appendLocalPropertyChangedToRevertibles(interval, propertyDeltas, revertibles);
+		});
+		collection.changeProperties(id, { foo: "two" });
+		containerRuntimeFactory.processAllMessages();
+
+		collection2.removeIntervalById(id);
+
+		containerRuntimeFactory.processAllMessages();
+
+		revertIntervalRevertibles(sharedString, revertibles.splice(0));
+		containerRuntimeFactory.processAllMessages();
+
+		assertIntervals(sharedString, collection, []);
+		assertIntervals(sharedString2, collection2, []);
+	});
+	it("remote interval remove interacting with reverting an interval remove", () => {
+		sharedString.insertText(0, "hello world");
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		const id = collection.add(0, 5, IntervalType.Simple).getIntervalId()!;
+		containerRuntimeFactory.processAllMessages();
+
+		collection.on("deleteInterval", (interval, local, op) => {
+			appendLocalDeleteToRevertibles(sharedString, interval, revertibles);
+		});
+		collection.removeIntervalById(id);
+		containerRuntimeFactory.processAllMessages();
+
+		collection2.removeIntervalById(id);
```

Comment:
Reverse the order of processAllMessages and remote delete so that the remote delete does something on client 2. With this order, client 2 see the client 1 delete before it locally deletes.